com.ibm.wala.cast.python.ml/source/com/ibm/wala/cast/python/ml/client/PythonTensorAnalysisEngine.java

```diff
@@ -131,47 +131,7 @@ private static Set<PointsToSetVariable> getDataflowSources(
         if (inst instanceof SSAAbstractInvokeInstruction) {
           // We potentially have a function call that generates a tensor.
           SSAAbstractInvokeInstruction ni = (SSAAbstractInvokeInstruction) inst;
-
-          if (ni.getCallSite()
-                  .getDeclaredTarget()
-                  .getName()
-                  .toString()
-                  .equals(TENSOR_GENERATOR_SYNTHETIC_FUNCTION_NAME)
-              && ni.getException() != vn) {
-            sources.add(src);
-            logger.info("Added dataflow source from tensor generator: " + src + ".");
-          } else if (ni.getNumberOfUses() > 1) {
-            // Get the invoked function from the PA.
-            int target = ni.getUse(0);
-            PointerKey targetKey =
-                pointerAnalysis.getHeapModel().getPointerKeyForLocal(localPointerKeyNode, target);
-
-            for (InstanceKey ik : pointerAnalysis.getPointsToSet(targetKey)) {
-              if (ik instanceof ConcreteTypeKey) {
-                ConcreteTypeKey ctk = (ConcreteTypeKey) ik;
-                IClass type = ctk.getType();
-                TypeReference reference = type.getReference();
-
-                if (reference.equals(NEXT.getDeclaringClass())) {
-                  // it's a call to `next()`. Look up the call to `iter()`.
-                  int iterator = ni.getUse(1);
-                  SSAInstruction iteratorDef = du.getDef(iterator);
-
-                  // Let's see if the iterator is over a tensor dataset.
-                  if (iteratorDef != null && iteratorDef.getNumberOfUses() > 1) {
-                    // Get the argument.
-                    int iterArg = iteratorDef.getUse(1);
-                    processInstructionInterprocedurally(
-                        iteratorDef, iterArg, localPointerKeyNode, src, sources, pointerAnalysis);
-                  } else
-                    // Use the original instruction. NOTE: We can only do this because `iter()` is
-                    // currently just passing-through its argument.
-                    processInstructionInterprocedurally(
-                        ni, iterator, localPointerKeyNode, src, sources, pointerAnalysis);
-                }
-              }
-            }
-          }
+          processInstruction(ni, du, localPointerKeyNode, src, vn, sources, pointerAnalysis);
```

Comment:
Maybe `processInvokeInstruction`?

Reply:
Sorry for the delay. Yes, I also initially thought of that. But, then I decided to use [method overloading](https://github.com/wala/ML/pull/201/files/02c2a6bd42521d24bf9552300ed8ce89a3f4fad8#diff-8c0e4df833a7e19f5c7b5e1ab40dc6a7e05277133c2641be105446d020bb4db7R215-R216):

```java
  private static boolean processInstruction(
      SSAAbstractInvokeInstruction instruction,
      ...
```

In other words, the first parameter's type implicitly informs that an invoke instruction is being "processed." I can change it back if desired.

Reply:
My preference is to change the method name to `processInvokeInstruction` for a bit more clarity (in my opinion).  But it's up to you; you read this code a lot more than I do 🙂 Not a huge deal

Reply:
I think that this is a good suggestion, but I wonder if we change this one, how many other ones are in a similar boat :).